### DISCUSSION TITLE
Query about project ideas

### DISCUSSION THREAD

**Discussion by tahsintunan** [cncf/mentoring#472] [Q&A] [answered]
From [this](https://github.com/kubernetes/community/blob/master/mentoring/programs/google-summer-of-code.md#google-summer-of-code) kubernetes/community doc, I got to know that Kubernetes is participating in the Google Summer of Code (GSoC) program under the Cloud Native Computing Foundation (CNCF) this year as well. [Here](https://github.com/kubernetes/community/blob/master/mentoring/programs/google-summer-of-code.md#contribute), they mention:

> As part of the application process, the Google Summer of Code program recommends that candidates make small contributions to the project they intend to apply for.

But to my understanding, the project ideas are yet to be submitted and finalized by the prospective mentors. So does it mean that we (prospective contributors) get to ideate/brainstorm over different project ideas as well? Or, do we have to wait till prospective mentors submit their ideas, and only then do we get to work on those ideas?

**Comment by Harshit2929** [comment]
I am also a newbie in open source i think they meant like after the actual project ideas are released [Feb14] and the time mentee applications starts[April 1- April 19],one has about 2 months to contribute in that particular project which is applying for gsoc.

**Comment by nate-double-u** [comment, marked as answer]
The window for CNCF Project Applications [has just opened](https://github.com/cncf/mentoring/discussions/469). Once the projects have submitted their ideas, we'll submit them to GSoC as a group.

As for ideating and brainstorming, please feel free to make comments on the PRs as they come in, or if you're involved in a project, bring up mentorship ideas with the maintainers.

**Comment by nate-double-u** [comment]
The [Google Summer of Code 2022 & CNCF](https://github.com/cncf/mentoring/discussions/565) discussion thread may be helpful now too.

**Comment by NikhilSharmaWe** [comment]
Just wanted to ask is there any specific template for the proposal for https://github.com/cncf/mentoring/blob/main/summerofcode/2022.md#new-golang-plugin-to-help-operator-authors-skill-up-size-medium project.

**Reply by nate-double-u** [reply]
I don't think there are any templates for proposals to the CNCF orgs, but the [GSoC Contributor/Student Guide](https://google.github.io/gsocguides/student/) has a section on [writing proposals](https://google.github.io/gsocguides/student/writing-a-proposal).

**Comment by NikhilSharmaWe** [comment]
Can we start writing our proposals ? Just confirming that all the projects in the project ideas are confirmed to be participated in GSoC.

**Reply by nate-double-u** [reply]
> Can we start writing our proposals ? 

Right now I'd suggest discussing application ideas with project mentors. On April 4 the GSoC contributor application period begins, and those applications are due on April 19. 

> Just confirming that all the projects in the project ideas are confirmed to be participated in GSoC.

CNCF is accepted as an org, but project ideas are not confirmed till after slots are requested. These slots are requested based on the GSoC contributor applications.


Here's the [Google Summer of Code 2022 Timeline](https://developers.google.com/open-source/gsoc/timeline)
